nemo/utils/exp_manager.py

```diff
@@ -346,8 +346,9 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
         else:
             check_resume(trainer, log_dir, cfg.resume_past_end, cfg.resume_ignore_no_checkpoint)
 
-    if cfg.resume_from_checkpoint is not None:
-        trainer.ckpt_path = cfg.resume_from_checkpoint
+    #  TODO: this behavior is undesirable, need ckpts in exp_dir to take priority if present over resume_from_checkpoint
+    # if cfg.resume_from_checkpoint is not None:
```

Comment:
@maanug-nv where are we taking care of the below lines then:

```
    if cfg.model.resume_from_checkpoint is not None:
        trainer.ckpt_path = cfg.model.resume_from_checkpoint
    logging.info(f'Resuming training from checkpoint: {trainer.ckpt_path}')
```
Since the pre training scripts were assigning the checkpoint to `trainer.ckpt_path` if we passed a checkpoint path for `resume_from_checkpoint` under model in config.

Reply:
Yes, so initially I moved those lines exactly as is to this place in `exp_manager.py`. After testing and discussing with @titu1994 , having those lines here (or in pretraining as they were before/are currently on main) has some undesirable behavior, details below. I wanted to keep this PR purely refactor (thought that would get it merged faster), so I'll correct the behavior in another PR. I can uncomment these lines if you prefer.

if 'resume_from_checkpoint' is set, that checkpoint is always used despite what is in the experiment dir. What makes more sense is that 'resume_from_checkpoint' is used if no exp_dir is present, but exp_dir takes priority if present.